gen/gotype.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package gen
+
+const (
+	// goTypeKey is a Thrift annotation that allows overriding the type of
+	// a typedef target type or a struct field type. By default, thrift set type
+	// corresponds to go map type. Following annotation on a typedef target type
+	// or struct field type, where the type is direct set type, causes thriftrw
+	// to generate go code with slice type instead of the default map type.
+	//
+	//     (go.type = "slice")
+	//
+	// Currently, only thrift set to go slice type overriding is supported.
+	goTypeKey = "go.type"
+	slice     = "slice"
```

Comment:
nit: change to "sliceType" please. The name "slice" is too generic for this shared namespace.

```suggestion
	sliceType = "slice"
```